File: gims_backend_api/src/main/java/com/kutub/InsuranceManagement/entity/utility/IssueOffice.java
package com.kutub.InsuranceManagement.entity.utility;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "issue_office")
public class IssueOffice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String name;

    private String address;
    private String phone;
    private String mobile;
    private String fax;
    private String email;
    private String website;
}

File: gims_mobile_app/lib/models/utility/issue_office.dart
class IssueOffice {
  final int? id;
  final String name;
  final String? address;
  final String? phone;
  final String? mobile;
  final String? fax;
  final String? email;
  final String? website;

  IssueOffice({
    this.id,
    required this.name,
    this.address,
    this.phone,
    this.mobile,
    this.fax,
    this.email,
    this.website,
  });

  factory IssueOffice.fromJson(Map<String, dynamic> json) {
    return IssueOffice(
      id: json['id'],
      name: json['name'],
      address: json['address'],
      phone: json['phone'],
      mobile: json['mobile'],
      fax: json['fax'],
      email: json['email'],
      website: json['website'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'address': address,
      'phone': phone,
      'mobile': mobile,
      'fax': fax,
      'email': email,
      'website': website,
    };
  }
}
